src/btree/bt_vrfy.c

```diff
@@ -686,20 +803,35 @@ __verify_tree(
             WT_RET(ret);
 
             WT_RET(bm->verify_addr(bm, session, unpack->data, unpack->size));
+
+            vs->cells_count_stack[my_stack_level] += vs->cells_count_stack[next_stack_level];
+            vs->total_sz_stack[my_stack_level] += vs->total_sz_stack[next_stack_level];
+            vs->key_sz_stack[my_stack_level] += vs->key_sz_stack[next_stack_level];
+            vs->val_sz_stack[my_stack_level] += vs->val_sz_stack[next_stack_level];
         }
         WT_INTL_FOREACH_END;
+        if (vs->dump_tree_shape)
+            printf("%s  =  children: %" PRIu64 "  cells: %" PRIu64 "  total_size: %" PRIu64
+                   "  keys_size: %" PRIu64 "  values_size: %" PRIu64 "\n",
+              __tree_stack(vs), vs->tree_stack[my_stack_level] + 1,
+              vs->cells_count_stack[my_stack_level], vs->total_sz_stack[my_stack_level],
+              vs->key_sz_stack[my_stack_level], vs->val_sz_stack[my_stack_level]);
         break;
     case WT_PAGE_ROW_INT:
         /* For each entry in an internal page, verify the subtree. */
         entry = 0;
+        next_stack_level = WT_MIN(vs->depth + 1, WT_ELEMENTS(vs->depth_internal) - 1);
         WT_INTL_FOREACH_BEGIN (session, page, child_ref) {
+            vs->tree_stack[my_stack_level] = entry++;
+            vs->cells_count_stack[next_stack_level] = vs->total_sz_stack[next_stack_level] =
+              vs->key_sz_stack[next_stack_level] = vs->val_sz_stack[next_stack_level] = 0;
+
             /*
              * It's a depth-first traversal: this entry's starting key should be larger than the
              * largest key previously reviewed.
              *
              * The 0th key of any internal page is magic, and we can't test against it.
              */
-            ++entry;
```

Comment:
Is it correct to remove this?

Reply:
It's moved over to the `vs->tree_stack[my_stack_level] = entry++;` line